lib/flux/app-state.js

```diff
@@ -700,19 +700,32 @@ export const actionMap = new ActionMap({
     loadPreferences: {
       creator({ callback, preferencesBucket }) {
         return dispatch => {
-          preferencesBucket.get(
-            'analytics_enabled',
-            (e, analyticsEnabled = { data: true }) => {
-              window.analyticsEnabled = analyticsEnabled.data;
-              if (typeof callback === 'function') callback();
+          const objectKey = 'preferences-key';
 
-              dispatch(
-                this.action('preferencesLoaded', {
-                  analyticsEnabled: analyticsEnabled.data,
-                })
-              );
+          preferencesBucket.get(objectKey, (e, preferences) => {
+            const preferencesData = preferences ? preferences.data : {};
+            const {
+              analytics_enabled: analyticsEnabled = true,
+            } = preferencesData;
+
+            // Create empty preferences object in the bucket if not yet there
+            if (preferences === undefined) {
+              preferencesBucket.update(objectKey, preferencesData);
```

Comment:
I added a few steps in here to make sure a `preferences-key` object is created if it doesn't exist yet.